### ISSUE TITLE
[NTI] java.lang.RuntimeException: INTERNAL COMPILER ERROR.

### ISSUE THREAD

**Issue by vobruba-martin** [google/closure-compiler#1952]
```javascript
navigator.serviceWorker.getRegistration('/app').then(function(/** !ServiceWorkerRegistration= */ swReg) {
	if (swReg) {
		/**
		 * Here should be warning because expected type is {?ServiceWorker}
		 * 
		 * @type {boolean}
		 */
		var swRegTmp = swReg.installing || swReg.waiting;
	
		if (swRegTmp) {
			swRegTmp.onstatechange = function() {
				if (swRegTmp.state === 'activated') {
					console.log(swReg.active);
				}
			};
		}
	}
});
```
```
media/js/test.js:12: WARNING - Property state never defined on swRegTmp of type Boolean
				if (swRegTmp.state === 'activated') {
				    ^

0 error(s), 1 warning(s)
java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

Can't getProp onstatechange of type boolean
In scope: swRegTmp.onstatechange() with root: FUNCTION  11 [length: 97] [originalname: swRegTmp.onstatechange] [source_file: media/js/test.js]
	at com.google.common.base.Preconditions.checkState(Preconditions.java:722)
	at com.google.javascript.jscomp.newtypes.JSType.getProp(JSType.java:1288)
	at com.google.javascript.jscomp.NewTypeInference.changeTypeIfFunctionNamespace(NewTypeInference.java:1128)
	at com.google.javascript.jscomp.NewTypeInference.createSummary(NewTypeInference.java:1104)
	at com.google.javascript.jscomp.NewTypeInference.analyzeFunction(NewTypeInference.java:765)
	at com.google.javascript.jscomp.NewTypeInference.process(NewTypeInference.java:414)
	at com.google.javascript.jscomp.PhaseOptimizer$NamedPass.process(PhaseOptimizer.java:285)
	at com.google.javascript.jscomp.PhaseOptimizer.process(PhaseOptimizer.java:217)
	at com.google.javascript.jscomp.Compiler.check(Compiler.java:847)
	at com.google.javascript.jscomp.Compiler.compileInternal(Compiler.java:742)
	at com.google.javascript.jscomp.Compiler.access$000(Compiler.java:83)
	at com.google.javascript.jscomp.Compiler$2.call(Compiler.java:690)
	at com.google.javascript.jscomp.Compiler$2.call(Compiler.java:687)
	at com.google.javascript.jscomp.CompilerExecutor$2.call(CompilerExecutor.java:93)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalStateException: Can't getProp onstatechange of type boolean
	... 18 more
```

**Comment by vobruba-martin** [comment]
https://closure-compiler-debugger.appspot.com/#input0%3Dnavigator.serviceWorker.getRegistration('%252Fapp').then(function(%252F**%2520!ServiceWorkerRegistration%253D%2520*%252F%2520swReg)%2520%257B%250A%2509if%2520(swReg)%2520%257B%250A%2509%2509%252F**%250A%2509%2509%2520*%2520Here%2520should%2520be%2520warning%2520because%2520expected%2520type%2520is%2520%257B%253FServiceWorker%257D%250A%2509%2509%2520*%2520%250A%2509%2509%2520*%2520%2540type%2520%257Bboolean%257D%250A%2509%2509%2520*%252F%250A%2509%2509var%2520swRegTmp%2520%253D%2520swReg.installing%2520%257C%257C%2520swReg.waiting%253B%250A%2509%250A%2509%2509if%2520(swRegTmp)%2520%257B%250A%2509%2509%2509swRegTmp.onstatechange%2520%253D%2520function()%2520%257B%250A%2509%2509%2509%2509if%2520(swRegTmp.state%2520%253D%253D%253D%2520'activated')%2520%257B%250A%2509%2509%2509%2509%2509console.log(swReg.active)%253B%250A%2509%2509%2509%2509%257D%250A%2509%2509%2509%257D%253B%250A%2509%2509%257D%250A%2509%257D%250A%257D)%253B%250A%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CHECK_TYPES_NEW_INFERENCE%3D1%26CLOSURE_PASS%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1

**Comment by dimvar** [comment]
Thanks for the report. Smaller repro:

    /** @type {boolean} */
    var b = true;
    if (b) {
      b.onstatechange = function() {};
    }
